Mathlib/CategoryTheory/Limits/Opposites.lean

```diff
@@ -401,112 +401,153 @@ theorem hasFiniteProducts_of_opposite [HasFiniteCoproducts Cᵒᵖ] : HasFiniteP
-  dsimp at this
-  rw [← Category.assoc, this]
-  simp only [colimit.cocone_x, Cofan.mk_pt, Equivalence.symm_functor, Discrete.natIsoFunctor,
-    comp_obj, op_obj, Iso.symm_hom, IsColimit.coconePointsIsoOfEquivalence_inv,
-    Equivalence.symm_inverse, Cocones.equivalenceOfReindexing_functor_obj, Iso.trans_inv,
-    Iso.symm_inv, isoWhiskerLeft_hom, Iso.trans_hom, whiskerLeft_comp, colimit.isColimit_desc,
-    colimit.ι_desc, Cocones.precompose_obj_pt, Cocone.whisker_pt, Cocones.precompose_obj_ι,
-    Cocone.whisker_ι, Category.assoc, NatTrans.comp_app, unop_op, Discrete.functor_obj,
-    const_obj_obj, Equivalence.invFunIdAssoc_inv_app, id_obj, op_map, Discrete.functor_map_id,
-    op_id, whiskerLeft_app, Discrete.natIso_hom_app, Iso.refl_hom, Cofan.mk_ι_app,
-    Discrete.opposite_functor_obj_as, Category.id_comp]
-  simp only [Discrete.functor, Function.comp_apply, id_eq, Discrete.opposite, Equivalence.mk,
-    id_obj, comp_obj, leftOp_obj, unop_op, Category.id_comp]
-
-lemma opProductIsoCoproduct_inv_comp_π_op {X : C} (π : (a : α) → X ⟶ Z a) :
-    (opProductIsoCoproduct Z).inv ≫ (Pi.lift π).op = Sigma.desc (fun a => Quiver.Hom.op (π a)) := by
-  rw [Iso.inv_comp_eq (opProductIsoCoproduct Z)]
-  congr
-  refine' Pi.hom_ext (f := Z) _ _ (fun a => _)
-  rw [Category.assoc, limit.lift_π, ← Quiver.Hom.unop_op (Pi.π Z a), ← unop_comp,
-    π_comp_opProductIsoCoproduct, ← unop_comp]
-  simp only [Fan.mk_pt, Fan.mk_π_app, colimit.ι_desc, Cofan.mk_pt, Cofan.mk_ι_app,
-    Quiver.Hom.unop_op]
+def Fan.op (f : Fan Z) : Cofan (fun a => op (Z a)) := Cofan.mk _ (fun a => (f.proj a).op)
+
+/-- If a `Fan` is limit, then its opposite is colimit. -/
+def Fan.IsLimit.op {f : Fan Z} (hf : IsLimit f) : IsColimit f.op := by
+  let e : Discrete.functor (fun a => Opposite.op (Z a)) ≅ (Discrete.opposite α).inverse ⋙
+    (Discrete.functor Z).op := Discrete.natIso (fun _ => Iso.refl _)
+  refine IsColimit.ofIsoColimit ((IsColimit.precomposeHomEquiv e _).2
+    (IsColimit.whiskerEquivalence hf.op (Discrete.opposite α).symm))
+    (Cocones.ext (Iso.refl _) (fun ⟨a⟩ => ?_))
+  dsimp
+  erw [Category.id_comp, Category.comp_id]
+  rfl
+
+/--
+The canonical isomorphism from the opposite of an abstract product to the corresponding coproduct
+in the opposite category.
+-/
+def opProductIsoCoproduct' {f : Fan Z} {c : Cofan (fun a => op (Z a))}
+    (hf : IsLimit f) (hc : IsColimit c) : op f.pt ≅ c.pt :=
+  IsColimit.coconePointUniqueUpToIso (Fan.IsLimit.op hf) hc
+
+variable (Z) in
+/--
+The canonical isomorphism from the opposite of the product to the coproduct in the opposite
+category.
+-/
+def opProductIsoCoproduct :
+    op (∏ Z) ≅ ∐ (fun z => op (Z z)) :=
+  opProductIsoCoproduct' (productIsProduct Z) (coproductIsCoproduct (fun z => op (Z z)))
+
+theorem π_comp_opProductIsoCoproduct'_hom {f : Fan Z} {c : Cofan (fun a => op (Z a))}
+    (hf : IsLimit f) (hc : IsColimit c) (b : α) :
+    (f.π.app ⟨b⟩).op ≫ (opProductIsoCoproduct' hf hc).hom = c.ι.app ⟨b⟩ := by
```

Comment:
```suggestion
    (f.π.app ⟨b⟩).op ≫ (opProductIsoCoproduct' hf hc).hom = c.ι.app ⟨b⟩ := by
```
For readibility, it would better to use `f.proj` and `c.inj` systematically.